pages/profile/index.js

```diff
@@ -42,6 +43,19 @@ module.exports = () => {
     next();
   });
 
+  app.use('/:profileId/permission', permissions('profile.permissions'),
+    (req, res, next) => {
+      Promise.resolve().then(() => {
+        if (req.user.profile.id === req.profileId) {
+          const err = new Error('Unauthorised');
+          err.status = 401;
+          next(err);
+        }
+      });
+      next();
+    },
```

Comment:
I'd move this code into the permission router as top-level middleware. That way all the logic for that page is encapsulated there.